src/main/java/core/basesyntax/controller/ControllerDaoImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.controller;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.Fruit;
+import java.util.List;
+
+public class ControllerDaoImpl implements ControllerDao<Fruit> {
+    private StorageDao<Fruit> storageDao = new StorageDaoImpl();
+
+    @Override
+    public boolean put(Fruit value) {
+        if (value == null) {
+            return false;
+        }
+        return storageDao.put(value);
+    }
+
+    @Override
+    public Fruit get(int index) {
+        if (storageDao.getAll().size() <= index
+                || index < 0
+                || storageDao.getAll().isEmpty()) {
```

Comment:
this check is redundant, because if you have list.size = 0 any index is bigger than size and it fits to first condition